lib/include/irritator/core.hpp

```diff
@@ -5221,74 +5221,77 @@ struct constant
 };
 
 struct filter {
+    static const size_t nPort=3;
     port x[1];
     port y[1];
     time sigma;
 
-    double default_lower_threshold;
-    double default_upper_threshold;
-
-    double lower_threshold;
-    double upper_threshold;
-
-    double default_value;
-    double default_input_coeff;
-
-    double value;
-    double input_coeff;
-
-    //filter() noexcept { 
-    //    default_lower_threshold=0.0;
-    //    default_upper_threshold = 1.0;
-    //    default_input_coeff=1.0;
-    //    default_value=0.0;
-
-    //}
+    double default_lower_threshold ;
+    double default_upper_threshold ;
+    double inValue ;
+    double defaultValues[nPort];
```

Comment:
I think this variable is useless. `default_low_threshold` and `default_upper_threshold`. I think you need:
```
double default_lower_threshold; // simulation parameter
double default_upper_threshold; // simulation parameter
double lower_threshold; // lower value during the simulation
double upper_threshold; // upper value during the simulation
double value; // message receives
```